webapp/src/components/StatusCheck.tsx

```diff
@@ -1,14 +1,44 @@
-import { Box, Typography } from "@mui/material";
+import { Box, capitalize, Stack, Typography } from "@mui/material";
 import noData from "assets/launch.svg";
 import React from "react";
 import { useParams } from "react-router-dom";
 import { getStatusEndpoint } from "services/api";
 import Loading from "./Loading";
+import Grid from "@mui/material/Grid";
+import Paper from "@mui/material/Paper";
+import CheckIcon from "./Icons/Check";
+import ErrorIcon from "@mui/icons-material/Error";
+import PendingOutlinedIcon from "@mui/icons-material/PendingOutlined";
 
 type Props = {
   children: React.ReactNode;
 };
 
+interface StringByString {
+  [key: string]: any;
+}
+
+const STATUS_ICON_MAPPING: StringByString = {
```

Comment:
The utility type [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) helps typing an object (dictionary) like that. Also, `React.ReactElement` would be a lot more robust that `any`. Finally, in the front end code we don't normally name things "mapping". For example, `SMART_TAG_FAMILY_ICONS` is just `ICONS` and not `ICON_MAPPING`. That's a detail, but I would make it uniform by renaming to `STATUS_ICONS`.
```suggestion
const STATUS_ICONS: Record<string, React.ReactElement> = {
```